newton/_src/geometry/broad_phase_common.py

```
@@ -37,6 +37,53 @@ def check_aabb_overlap(
+        return check_aabb_overlap(
+            box_lower[shape1], box_upper[shape1], cutoff1, box_lower[shape2], box_upper[shape2], cutoff2
+        )
+
+    cutoff_combined = cutoff1 + cutoff2
+    relative_sweep = shape_sweep[shape1] - shape_sweep[shape2]
+    lower_box1 = box_lower[shape1]
+    upper_box1 = box_upper[shape1]
+    lower_box2 = box_lower[shape2]
+    upper_box2 = box_upper[shape2]
+    enter = float(0.0)
+    exit = float(1.0)
```

Ruff (0.16.1)

[error] Variable `exit` is shadowing a Python builtin

(A001)